controllers/helmchart_controller.go

```diff
@@ -493,9 +493,17 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
+	version := helmChart.Metadata.Version
+	// Uses GitRepository or Bucket revision with chart version if ignoring Chart.yaml version
+	if chart.Spec.IgnoreVersion {
+		re0 := regexp.MustCompile("-flux+.*")
+		re1 := regexp.MustCompile(".*/")
+		version = fmt.Sprintf("%s-flux+%s", re0.ReplaceAllString(version, ""), re1.ReplaceAllString(artifact.Revision, ""))
```

Comment:
@hiddeco I was thinking `0.1.0-flux+sha` would be the right pattern for chart. I like how it's clear in a `helm ls` that this chart is managed by Flux. Thoughts?

Reply:
I think having a `flux` suffix is OK, but it should be part of the string after the `+`, as `-flux` makes everything a pre-release.

I would also use the SemVer library to add the metadata, if you parse the version using https://github.com/fluxcd/pkg/blob/main/version/version.go#L27, you get an object back that should allow you to make modifications to the SemVer and add metadata: https://github.com/Masterminds/semver/blob/v3.1.1/version.go#L349